File: template-service/src/main/kotlin/hu/gde/kerdezz/templateservice/domain/Survey.kt
package hu.gde.kerdezz.templateservice.domain

import org.springframework.data.mongodb.core.mapping.Document
import java.time.Instant
import java.time.LocalDate

@Document("surveyTemplates")
data class Survey(
  val id: String?,
  val user: String,
  val createdAt: Instant = Instant.now(),
  val name: String,
  val text: String?,
  val isAnonymous: Boolean,
  val isMultiple: Boolean,
  val visibility: Visibility,
  val questions: List<Question>,
  val allowedEmails: List<String>?
)

data class Question(
  val id: String,
  val required: Boolean = false,
  val type: QuestionType,
  val text: String,
  val options: List<String>?,
  val min: Int?,
  val max: Int?,
  val minDate: LocalDate?,
  val maxDate: LocalDate?
)

File: kerdezz-frontend/src/app/domain/survey.ts
export class Survey {
  id?: string;
  name: string;
  anonymous: boolean;
  multiCompletion: boolean;
  visibility: string;
  questions: Question[];
  allowedEmails: string[] = [];
}

export interface Question {
  id?: string;
  questionText?: string;
  questionType?: string;
  required: boolean;
  options: Option[];
  minValue?: number;
  maxValue?: number;
  minDate?: Date;
  maxDate?: Date;
}

export interface Option {
  value: string
}
